File: src/main/java/pl/training/module3/adapter/ConsoleQuoteNotifier.java
package pl.training.module3.adapter;

import java.util.Objects;

import pl.training.module3.application.QuoteNotifier;
import pl.training.module3.domain.DeliveryQuote;

public final class ConsoleQuoteNotifier implements QuoteNotifier {
    @Override
    public void quoteCreated(DeliveryQuote quote) {
        Objects.requireNonNull(quote);

        System.out.printf(
                "Quote ready for %s: %s costs %s%n",
                quote.customerEmail(),
                quote.method(),
                quote.price());
    }
}

File: typescript/src/module3/adapter/ConsoleQuoteNotifier.ts
import { requireNonNull } from '../../shared/requireNonNull.js';
import type { QuoteNotifier } from '../application/QuoteNotifier.js';
import type { DeliveryQuote } from '../domain/DeliveryQuote.js';

export class ConsoleQuoteNotifier implements QuoteNotifier {
  quoteCreated(quote: DeliveryQuote): void {
    requireNonNull(quote);

    console.log(
      `Quote ready for ${quote.customerEmail}: ${quote.method} costs ${quote.price.toFixed(2)}`);
  }
}
